From 2e2d1098113e6c96e296acb1a503a82298e8e7d0 Mon Sep 17 00:00:00 2001
From: Chen Wandun <chenwandun@huawei.com>
Date: Tue, 16 Aug 2022 21:10:23 +0800
Subject: [PATCH] Revert "mm/page_cache_limit: do shrink_page_cache when adding
 page to page cache"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2e2d1098113e6c96e296acb1a503a82298e8e7d0
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56I4P
CVE: NA
backport: openEuler-22.03-LTS

--------------------------------

This reverts commit 9fea105d74885fef299b43e6734d19ba8921242e.

Signed-off-by: Chen Wandun <chenwandun@huawei.com>
Reviewed-by: Tong Tiangen <tongtiangen@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/page_cache_limit.h | 2 --
 include/linux/pagemap.h          | 2 --
 mm/filemap.c                     | 2 --
 3 files changed, 6 deletions(-)

diff --git a/include/linux/page_cache_limit.h b/include/linux/page_cache_limit.h
index 2df08a0604d8..7906b12af947 100644
--- a/include/linux/page_cache_limit.h
+++ b/include/linux/page_cache_limit.h
@@ -19,11 +19,9 @@ unsigned long __shrink_node_page_cache(int nid, gfp_t mask,
 		unsigned long nr_to_reclaim, enum page_cache_reclaim_flag flag);
 void kpagecache_limitd_stop(int nid);
 int kpagecache_limitd_run(int nid);
-void wakeup_all_kpagecache_limitd(void);
 #else
 static inline void kpagecache_limitd_stop(int nid) {}
 static inline int kpagecache_limitd_run(int nid) { return 0; }
-static inline void wakeup_all_kpagecache_limitd(void) {}
 #endif
 
 #endif
diff --git a/include/linux/pagemap.h b/include/linux/pagemap.h
index dbb25f1dc2e9..0bfa9cce6589 100644
--- a/include/linux/pagemap.h
+++ b/include/linux/pagemap.h
@@ -15,7 +15,6 @@
 #include <linux/bitops.h>
 #include <linux/hardirq.h> /* for in_interrupt() */
 #include <linux/hugetlb_inline.h>
-#include <linux/page_cache_limit.h>
 
 struct pagevec;
 
@@ -778,7 +777,6 @@ static inline int add_to_page_cache(struct page *page,
 {
 	int error;
 
-	wakeup_all_kpagecache_limitd();
 	__SetPageLocked(page);
 	error = add_to_page_cache_locked(page, mapping, offset, gfp_mask);
 	if (unlikely(error))
diff --git a/mm/filemap.c b/mm/filemap.c
index edb94663c5df..98b448d9873f 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -42,7 +42,6 @@
 #include <linux/psi.h>
 #include <linux/ramfs.h>
 #include <linux/page_idle.h>
-#include <linux/page_cache_limit.h>
 #include "internal.h"
 
 #define CREATE_TRACE_POINTS
@@ -924,7 +923,6 @@ int add_to_page_cache_lru(struct page *page, struct address_space *mapping,
 	void *shadow = NULL;
 	int ret;
 
-	wakeup_all_kpagecache_limitd();
 	__SetPageLocked(page);
 	ret = __add_to_page_cache_locked(page, mapping, offset,
 					 gfp_mask, &shadow);
-- 
2.34.1

